scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -912,6 +976,35 @@ void BPF_STRUCT_OPS(lavd_running, struct task_struct *p)
 	try_proc_introspec_cmd(p, taskc);
 }
 
+void BPF_STRUCT_OPS(lavd_tick, struct task_struct *p)
+{
+	struct cpu_ctx *cpuc;
+	struct task_ctx *taskc;
+	u64 now;
+
+	/*
+	 * Update task statistics
+	 */
+	cpuc = get_cpu_ctx_task(p);
+	taskc = get_task_ctx(p);
+	if (!cpuc || !taskc) {
+		scx_bpf_error("Failed to lookup context for task %d", p->pid);
+		return;
+	}
+
+	now = scx_bpf_now();
+	account_task_runtime(p, taskc, cpuc, now);
+
+	/*
+	 * If this task is slice-boosted and there is a pinned task that
```

Comment:
One other idea is maybe if the task is a nice take it could just yield, but I guess preemption handling is done elsewhere.